test/integration/server.cc

```diff
@@ -220,6 +223,9 @@ void IntegrationTestServerImpl::createAndRunEnvoyServer(
     Random::RandomGeneratorPtr&& random_generator, ProcessObjectOptRef process_object,
     Buffer::WatermarkFactorySharedPtr watermark_factory) {
   {
+    if (skipAsserts()) {
+      skip = std::make_unique<Thread::SkipAsserts>();
```

Comment:
still not sure why you need to instantiate this; it's just got one static method.